app/helpers/sellers_helper.rb

```diff
@@ -4,9 +4,22 @@ module SellersHelper
   def self.generate_seller_json(seller:)
     locations = seller.locations
     distributor = seller.distributor
+    fees = seller.fees
+
     json = seller.as_json
     json['distributor'] = distributor.as_json unless distributor.nil?
     json['locations'] = locations.as_json
+    json['fees'] = fees.as_json
+
+    # Take into account the associated fees when calculating the cost per meal
+    # for the customer. eg) We might generate a $10 gift card for GAM, but if
+    # there is a 10% fee associated with creating that gift card, then we should
+    # charge the customer $11.
+    if seller.cost_per_meal.present?
+      json['cost_per_meal'] = seller.cost_per_meal + fees.inject(0) do |_total, fee|
+        (seller.cost_per_meal * fee.multiplier).ceil
+      end
+    end
```

Comment:
I just thought about this — should build fees such that < 1 it's a discount, and > 1 it costs more? So if a multiplier is .1 on a $10 meal it'll be $1, and if it's 1.1 on a $10 meal it'll be $11?

Right now it's built such that it only can add cost to a meal and can't take away.

What do y'all think @danthaman44 @jacoblb64

Reply:
Actually, I think I might keep it as is. If we want to have discounts, I think we can model them as a separate object called Discounts. It just makes it clearer that way